syntax = "proto3";
package report_generator;

message GenerateReportRequest {
  string analyzed_data_path = 1;
  string report_format = 2; // csv, html, json
}

message GenerateReportResponse {
  bool success = 1;
  string message = 2;
  string report_path = 3;
  string report_summary = 4;
}

service ReportGeneratorService {
  rpc GenerateReport(GenerateReportRequest) returns (GenerateReportResponse);
}